File: static/CWE-22/java/vulnerable.java
// CWE-22: Path Traversal
// A user-supplied filename is used to build a filesystem path without validation.
// Vulnerable sink: Files.readAllBytes(Paths.get(userInput))

import java.io.File;
import java.io.FileInputStream;
import java.nio.file.Files;
import java.nio.file.Paths;

class CWE_22 {
    public byte[] readFile(String userInput) throws Exception {
        File f = new File("/var/www/uploads", userInput);
        FileInputStream in = new FileInputStream(f);
        in.close();
        return Files.readAllBytes(Paths.get(userInput));
    }
}

File: static/CWE-22/csharp/vulnerable.cs
// CWE-22: Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal')
// User-controlled filename is combined with a base directory and read without validation.
// Vulnerable sink: File.ReadAllText(Path.Combine(baseDir, userInput))

using System.IO;

namespace GuineaPig.Sast
{
    public class CWE_22
    {
        public string ReadFile(string userInput)
        {
            string baseDir = "/var/www/files";
            string fullPath = Path.Combine(baseDir, userInput);
            return File.ReadAllText(fullPath);
        }
    }
}
